### PR TITLE
PR: Create generic switcher

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)
* [x] Included a screenshot (if affecting the UI)

switcher

s

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

**Comment by CAM-Gerlach** [comment]
@goanpeca Let me know when this is ready for a first round of UI/UX/functional testing and review. Thanks!

**Comment by goanpeca** [comment]
Will do! In theory with the basic example I added you can already do some basic tests.

This PR will not hook to the existinh switcher logic cause otherwise it will become too big.

Also tests are missing

**Comment by goanpeca** [comment]
Hi @dalthviz I still need to provide more feedback but you can start working, by making the examples within the file work with the dark style. I hardcoded some values to make it work.

The Widget and items should receive the styles maybe as a dict and not query the CONF system directly so that the widget is decoupled. If no styles are provided, the widget can the query the palette to get the basic styles. In theory this should also work with QDarkStyle when we implement the changes to colors on the palette and not only as a stylesheet (see https://github.com/ColinDuquesnoy/QDarkStyleSheet/issues/171)

**Comment by ccordoba12** [comment]
@dalthviz, this is looking pretty nice! Please add some tests to the new switcher so we can merge this one.

File: spyder/plugins/editor/widgets/recover.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 from qtpy.QtWidgets import (QDialog, QDialogButtonBox, QHBoxLayout, QLabel,
                             QMessageBox, QPushButton, QTableWidget,
                             QVBoxLayout, QWidget)
+from spyder.py3compat import to_text_string
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/recover.py]: This import is not used now, Please remove it.

**Comment by goanpeca** [comment]
Thanks for the work finishing this @dalthviz !

**Pull request merged**